kairon/shared/channels/mail/processor.py

```
@@ -0,0 +1,297 @@
+    def get_mail_channel_state_data(bot):
+        """
+        Get mail channel state data
+        """
+        try:
+            state = MailChannelStateData.objects(bot=bot).first()
+            if not state:
+                state = MailChannelStateData(bot=bot)
+                state.save()
+            return state
+        except Exception as e:
+            raise AppException(str(e))
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)